Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.Common/Providers/GraphUtility.cs

```diff
@@ -155,32 +148,56 @@ public class GraphUtility : IGraphUtility
                 { "CallingAssembly", Assembly.GetCallingAssembly().GetName().Name },
             };
 
+            List<ShiftTeams> shiftTeams = new List<ShiftTeams>();
+            var hasMoreTeams = false;
+
             this.telemetryClient.TrackTrace(BusinessLogicResource.FetchShiftTeamDetailsAsync, fetchShiftUserDetailsProps);
             var hcfClient = this.httpClientFactory.CreateClient("GraphBetaAPI");
             hcfClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
             hcfClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            using (var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "groups")
-            {
-            })
+            var requestUri = "groups";
```

Comment:
Do you need all **groups**, including groups that are not teams?
If you only need teams, see: https://docs.microsoft.com/en-us/graph/teams-list-all-teams

Reply:
@aosolis , Added filter while fetching teams as suggested by you

Reply:
@PinkeyJRatnani Did you test this? The filter is wrong.
https://docs.microsoft.com/en-us/graph/teams-list-all-teams#get-a-list-of-groups-using-beta-apis

You have this:
"groups?$filter=resourceProvisioningOptions"
which is not the same thing as this:
GET /groups?$filter=resourceProvisioningOptions/Any(x:x eq 'Team')